analyses/cell-type-ewings/exploratory_analysis/annotation_notebooks/SCPCS000492/SCPCL000824_tumor-gene-expression-exploration.Rmd

```diff
@@ -0,0 +1,749 @@
+The thicker line indicates the chosen threshold by the algorithm. 
+This corresponds to the highest value that reduces the false positives. 
+
+Here we can see a bimodal distribution of all gene sets except `SILIGAN`, which is to be expected based on the original distribution plots. 
+
+For the remaining analysis we will use the results from running `AUCell` with only the marker gene list. 
+Let's compare the assignments from our validated classifications (determined by taking the consensus between `InferCNV`, and `CopyKAT` for `SCPCL000822`) to using `AUCell` with `SCPCL000822`. 
+
+```{r}
+# compare AUCell classification vs cnv classification in ref
+# pull out tumor cells found by AUCell
+tumor_cells <- ref_assignment$markers$assignment
+
+ref_classification_df <- ref_classification_df |> 
+  # add new column with auc classification 
+  dplyr::mutate(auc_classification = dplyr::if_else(barcodes %in% tumor_cells, "Tumor", "Normal"))
+
+# filter out any ambiguous cells 
+caret_df <- ref_classification_df |> 
+  dplyr::filter(tumor_cell_classification != "Ambiguous")
+
+# compare using a confusion matrix
+caret::confusionMatrix(
+    table(
+      caret_df$tumor_cell_classification, 
+      caret_df$auc_classification)
+  ) 
+```
+
+It looks like we get pretty good agreement between our original classification and the updated `AUCell` classification using the marker gene list.
+
+Now we repeat on `SCPCL000824`. 
+
+```{r}
+# get new auc and plot thresholds 
+new_auc <- AUCell::AUCell_run(counts(sce), collection)
+new_assignment <- AUCell::AUCell_exploreThresholds(new_auc, assign = TRUE)
+```
+
+Here we do not see any bimodal distributions and that most cells are considered to not be expressing the specified gene sets, which doesn't fit with our previous findings. 
+I expect that without a bimodal distribution this model does not do a good job of calculating a threshold. 
+
+```{r}
+ref_auc_df <- ref_auc@assays@data$AUC |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("gene_set") |> 
+  tidyr::pivot_longer(!gene_set, 
+                      names_to = "barcodes", 
+                      values_to = "auc") |> 
+  #dplyr::filter(gene_set == "markers") |> 
+  dplyr::mutate(sample = "SCPCL000822")
+
+new_auc_df <- new_auc@assays@data$AUC |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("gene_set") |> 
+  tidyr::pivot_longer(!gene_set, 
+                      names_to = "barcodes", 
+                      values_to = "auc") |> 
+  #dplyr::filter(gene_set == "markers") |> 
+  dplyr::mutate(sample = "SCPCL000824")
+
+all_auc_df <- dplyr::bind_rows(list(ref_auc_df, new_auc_df))
+
+ggplot(all_auc_df, aes(x = auc, color = sample)) +
+  geom_density() +
+  facet_grid(rows = vars(gene_set))
+```
+
+Looking at the AUC across both samples for all gene sets, we see that they are in the same range of values and tend to mirror each other. 
+The main difference here is the lack of bimodal distribution in `SCPCL000824`. 
```

Comment:
I wouldn't necessarily say they mirror each other... The peak for 824 looks to be in between the two peaks for 822 in many cases, with the marker gene list being the exception where the peaks do line up. Which is a good argument that the marker gene list is a good set to use.